File: 0009-palindrome-number/0009-palindrome-number.py
class Solution:
    def isPalindrome(self, x: int) -> bool:
        nb = str(x)
        p1 = 0
        p2 = len(nb) - 1
        
        if x < 0:
            return False
        
        while p1 < p2:
            if nb[p1] != nb[p2]:
                return False
            p1 += 1
            p2 -= 1
        
        return True

File: 0009-palindrome-number/0009-palindrome-number.java
class Solution {
    public boolean isPalindrome(int x) {
        String nb = Integer.toString(x);
        int p1 = 0;
        int p2 = nb.length() - 1;
        
        if(x < 0){
            return false;
        }
        
        while(p1 < p2){
            if(nb.charAt(p1) != nb.charAt(p2)){
                return false;
            }
            p1++;
            p2--;
        }
        
        return true;
    }
}